### PR TITLE
implement `Vec<more_primitive_types>` and `ZeroCopyBuffer<Vec<more_primitive_types>>` - originally only support few types like `Vec<u8>`/`ZeroCopyBuffer<Vec<u8>>`

### PR DESCRIPTION
**Pull request by fzyzcjy**

### CODE REVIEW

**Comment by fzyzcjy** [comment]
@shekohex Hi could you please have a look at this PR

**Comment by fzyzcjy** [comment]
@shekohex Hi could you please have a look at this PR? The upstream PR (https://github.com/fzyzcjy/flutter_rust_bridge/pull/162) is blocked by this one. If this lib is not released soon, then I have to fork and release a cloned package on crates.io ... Thanks!

**Comment by shekohex** [comment]
I will give it a review today don't worry, and publish a new version by the end of the day.

Thank you.

**Comment by fzyzcjy** [comment]
You are welcome! Looking forward to the new version

File: src/into_dart.rs [hunk 1]
```diff
@@ -92,43 +109,89 @@ impl IntoDart for CString {
     }
 }
 
-trait Num8Bit {}
+impl IntoDartExceptPrimitive for CString {}
 
-impl Num8Bit for u8 {}
+pub(crate) trait DartTypedDataTypeVisitor {
+    fn visit<T: DartTypedDataTypeTrait>(&self);
+}
```
Comment (shekohex) [inline src/into_dart.rs:116]: Could you please explain (adding more comments to the code) about this trait? like why do we need it?

File: src/into_dart.rs [hunk 2]
```diff
@@ -92,43 +109,89 @@ impl IntoDart for CString {
...
-        length: vec.len() as isize,
-        values: vec.as_mut_ptr() as *mut _,
-    };
-    DartCObject {
-        ty: DartCObjectType::DartTypedData,
-        value: DartCObjectValue {
-            as_typed_data: data,
-        },
-    }
+    fn function_pointer_of_free_zero_copy_buffer() -> DartHandleFinalizer;
```
Comment (shekohex) [inline src/into_dart.rs:122]: should it be just `finalizer`? easier to pronounce, I guess 😅

**Review by shekohex — CHANGES_REQUESTED** [review]
Looks good to so far, just few changes and we are good to go!

File: src/into_dart.rs (hunk 2 above)
Comment (fzyzcjy) [reply src/into_dart.rs:122]: Maybe no, since it is finalizer **of ZeroCopyBuffer**, not finalizer of the type T.

**Comment by fzyzcjy** [comment]
comments added.

**Review by shekohex — APPROVED** [review]

**Comment by shekohex** [comment]
Perfect! Will publish a new version in the morning.

> Sorry it's too late right now (2:30 AM).

Thank you for your contribution to the project!

**Comment by fzyzcjy** [comment]
🎉  You are welcome!

> Will publish a new version in the morning.

Looking forward to it!

btrw now it is 8:30 AM here - so that is why I did not reply to your questions hours ago

**Comment by shekohex** [comment]
Published `v0.1.11` 🎉

**Comment by fzyzcjy** [comment]
:tada:

**Pull request merged**